rdr_service/genomic/genomic_job_controller.py

```diff
@@ -1471,5 +1468,44 @@ def get_report(self, **kwargs):
             )
 
         self.job_run_result = GenomicSubProcessResult.SUCCESS
-
         return report_result
+
+    def send_validation_emails(self):
+        """
+        Send emails via sendgrid for previous day
+        validation failures on GC/DRC manifest ingestions
+        """
+        email_config = config.getSettingJson(config.GENOMIC_DAILY_VALIDATION_EMAILS, {})
+
+        if not email_config.get('send_emails'):
+            return
+
+        validation_incidents = self.incident_dao.get_new_ingestion_incidents()
+
+        if not validation_incidents:
+            logging.warning('No records found for validation email notifications')
+            return
+
+        recipients = email_config.get('recipients')
+
+        for gc, recipient_list in recipients.items():
+            gc_validation_emails_to_send = list(filter(lambda x: x.submitted_gc_site_id == gc, validation_incidents))
+
+            if gc_validation_emails_to_send:
+                for gc_validation_email in gc_validation_emails_to_send:
+                    validation_message = gc_validation_email.message.split(':')[1]
+                    message = f"{validation_message}\n"
+                    message += f"Full file path: gs://{gc_validation_email.filePath}\n"
+                    message += f"Please correct this file, gs://{gc_validation_email.filePath}, and re-upload to " \
+                               f"designated bucket."
+
+                    email_message = Email(
+                        recipients=recipient_list,
+                        subject="All of Us GC/DRC Manifest Ingestion Failure",
+                        plain_text_content=message
+                    )
+
+                    EmailService.send_email(email_message)
```

Comment:
(possible minor optimization) Can one email be sent that lists all recipients (bcc if they shouldn't see that the others got the email as well), rather than send separate emails for each of them?

Reply:
So each GC should have their own email sent to each of their recipients for the incidents created. Since there are 3 different GCs that have no relation to each other, we have to send separate emails.

Reply:
Ah, my mistake. I had misread it the first time through and thought there was a for-loop on the recipients. Looks good!